relay-dynamic-config/src/project.rs

```diff
@@ -88,6 +88,9 @@ pub struct ProjectConfig {
     /// relays that might still need them.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub span_description_rules: Option<Vec<SpanDescriptionRule>>,
+    /// Configuration for filtering metrics.
+    #[serde(default, skip_serializing_if = "skip_deny_metrics")]
```

Comment:
I'd rather make this `Option<ErrorBoundry<...>>`

Reply:
well, atm it's just a vector underneath, and we don't really do Option<Vec> in the projectconfig if you check out the other values. I'm not sure what the difference between `None` and an empty vec would be, so I think we might as well not encode that.

Reply:
`Metrics` is an Object not a Vector? If you refer to the fields it contains, that does and should not matter.